### ISSUE TITLE
[Bug/Support]: Failed dependency (git-commit) when installing magit

### ISSUE THREAD

**Issue by fabcontigiani** [progfolio/elpaca#285]
### Confirmation

- [X] I have checked the documentation (README, Wiki, docstrings, etc)
- [ ] I am  checking these without reading them.
- [X] I have searched previous issues to see if my question is a duplicate.

### Elpaca Version

Elpaca dc66af6 grafted, HEAD -> master, origin/master, origin/HEAD
installer:      0.7
emacs-version:  GNU Emacs 29.3 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.41, cairo version 1.18.0)
git --version:  git version 2.44.0Elpaca dc66af6 grafted, HEAD -> master, origin/master, origin/HEAD
installer:      0.7
emacs-version:  GNU Emacs 29.3 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.41, cairo version 1.18.0)
git --version:  git version 2.44.0

### Operating System

Arch Linux
```
$ uname -a
Linux archlinux 6.8.2-arch2-1 #1 SMP PREEMPT_DYNAMIC Thu, 28 Mar 2024 17:06:35 +0000 x86_64 GNU/Linux
```

### Description

Hi! I'm getting an error similar to #265 when installing magit, except for git-commit, all the other dependencies seems to install correctly:
![image](https://github.com/progfolio/elpaca/assets/25231400/79648bb4-82cb-4524-96a3-5b82cc229315)

Relevant configuration used:
```elisp
(elpaca elpaca-use-package
  (elpaca-use-package-mode)
  (setq use-package-always-ensure t))

(elpaca-wait)

(use-package magit)
```

`M-x elpaca-info` for `magit`:
```
magit [MELPA|NonGNU-devel ELPA]
A Git porcelain inside Emacs.

source: MELPA
url: https://github.com/magit/magit
menu item recipe:
( :package "magit"
  :fetcher github
  :repo "magit/magit"
  :files ("lisp/magit*.el"
	  "lisp/git-rebase.el" "docs/magit.texi"
	  "docs/AUTHORS.md" "LICENSE"
	  "Documentation/magit.texi"
	  "Documentation/AUTHORS.md" (:exclude
				      "lisp/magit-libgit.el"
				      "lisp/magit-libgit-pkg.el"
				      "lisp/magit-section.el"
				      "lisp/magit-section-pkg.el"))
  :source "MELPA")
full recipe:
( :package "magit"
  ;; Inherited from elpaca-order-functions.
  :depth 1
  :inherit t
  :protocol https
  ;; Inherited from elpaca-menu-item.
  :source "MELPA"
  :files ("lisp/magit*.el"
	  "lisp/git-rebase.el" "docs/magit.texi"
	  "docs/AUTHORS.md" "LICENSE"
	  "Documentation/magit.texi"
	  "Documentation/AUTHORS.md" (:exclude
				      "lisp/magit-libgit.el"
				      "lisp/magit-libgit-pkg.el"
				      "lisp/magit-section.el"
				      "lisp/magit-section-pkg.el"))
  :repo "magit/magit"
  :fetcher github)
dependencies: 
  emacs         >= 25.1
  compat        >= 29.1.4.4
  dash          >= 20240103
  git-commit    >= 20240123
  magit-section >= 20240114
  seq           >= 2.24
  transient     >= 20240201
  with-editor   >= 20240101
dependents: nil
installed version: 3.3.0.50-git b5637d6
statuses:
  (failed blocked ref-checked-out queued)
files:
  $REPOS/magit/LICENSE                       ! $BUILDS/magit/LICENSE
  $REPOS/magit/docs/AUTHORS.md               ! $BUILDS/magit/AUTHORS.md
  $REPOS/magit/docs/magit.texi               ! $BUILDS/magit/magit.texi
  $REPOS/magit/lisp/git-rebase.el            ! $BUILDS/magit/git-rebase.el
  $REPOS/magit/lisp/magit-apply.el           ! $BUILDS/magit/magit-apply.el
  $REPOS/magit/lisp/magit-autorevert.el      ! $BUILDS/magit/magit-autorevert.el
  $REPOS/magit/lisp/magit-base.el            ! $BUILDS/magit/magit-base.el
  $REPOS/magit/lisp/magit-bisect.el          ! $BUILDS/magit/magit-bisect.el
  $REPOS/magit/lisp/magit-blame.el           ! $BUILDS/magit/magit-blame.el
  $REPOS/magit/lisp/magit-bookmark.el        ! $BUILDS/magit/magit-bookmark.el
  $REPOS/magit/lisp/magit-branch.el          ! $BUILDS/magit/magit-branch.el
  $REPOS/magit/lisp/magit-bundle.el          ! $BUILDS/magit/magit-bundle.el
  $REPOS/magit/lisp/magit-clone.el           ! $BUILDS/magit/magit-clone.el
  $REPOS/magit/lisp/magit-commit.el          ! $BUILDS/magit/magit-commit.el
  $REPOS/magit/lisp/magit-core.el            ! $BUILDS/magit/magit-core.el
  $REPOS/magit/lisp/magit-diff.el            ! $BUILDS/magit/magit-diff.el
  $REPOS/magit/lisp/magit-ediff.el           ! $BUILDS/magit/magit-ediff.el
  $REPOS/magit/lisp/magit-extras.el          ! $BUILDS/magit/magit-extras.el
  $REPOS/magit/lisp/magit-fetch.el           ! $BUILDS/magit/magit-fetch.el
  $REPOS/magit/lisp/magit-files.el           ! $BUILDS/magit/magit-files.el
  $REPOS/magit/lisp/magit-git.el             ! $BUILDS/magit/magit-git.el
  $REPOS/magit/lisp/magit-gitignore.el       ! $BUILDS/magit/magit-gitignore.el
  $REPOS/magit/lisp/magit-log.el             ! $BUILDS/magit/magit-log.el
  $REPOS/magit/lisp/magit-margin.el          ! $BUILDS/magit/magit-margin.el
  $REPOS/magit/lisp/magit-merge.el           ! $BUILDS/magit/magit-merge.el
  $REPOS/magit/lisp/magit-mode.el            ! $BUILDS/magit/magit-mode.el
  $REPOS/magit/lisp/magit-notes.el           ! $BUILDS/magit/magit-notes.el
  $REPOS/magit/lisp/magit-patch.el           ! $BUILDS/magit/magit-patch.el
  $REPOS/magit/lisp/magit-pkg.el             ! $BUILDS/magit/magit-pkg.el
  $REPOS/magit/lisp/magit-process.el         ! $BUILDS/magit/magit-process.el
  $REPOS/magit/lisp/magit-pull.el            ! $BUILDS/magit/magit-pull.el
  $REPOS/magit/lisp/magit-push.el            ! $BUILDS/magit/magit-push.el
  $REPOS/magit/lisp/magit-reflog.el          ! $BUILDS/magit/magit-reflog.el
  $REPOS/magit/lisp/magit-refs.el            ! $BUILDS/magit/magit-refs.el
  $REPOS/magit/lisp/magit-remote.el          ! $BUILDS/magit/magit-remote.el
  $REPOS/magit/lisp/magit-repos.el           ! $BUILDS/magit/magit-repos.el
  $REPOS/magit/lisp/magit-reset.el           ! $BUILDS/magit/magit-reset.el
  $REPOS/magit/lisp/magit-sequence.el        ! $BUILDS/magit/magit-sequence.el
  $REPOS/magit/lisp/magit-sparse-checkout.el ! $BUILDS/magit/magit-sparse-checkout.el
  $REPOS/magit/lisp/magit-stash.el           ! $BUILDS/magit/magit-stash.el
  $REPOS/magit/lisp/magit-status.el          ! $BUILDS/magit/magit-status.el
  $REPOS/magit/lisp/magit-submodule.el       ! $BUILDS/magit/magit-submodule.el
  $REPOS/magit/lisp/magit-subtree.el         ! $BUILDS/magit/magit-subtree.el
  $REPOS/magit/lisp/magit-tag.el             ! $BUILDS/magit/magit-tag.el
  $REPOS/magit/lisp/magit-transient.el       ! $BUILDS/magit/magit-transient.el
  $REPOS/magit/lisp/magit-wip.el             ! $BUILDS/magit/magit-wip.el
  $REPOS/magit/lisp/magit-worktree.el        ! $BUILDS/magit/magit-worktree.el
  $REPOS/magit/lisp/magit.el                 ! $BUILDS/magit/magit.el
log:
  [2024-04-03 13:09:28] Package queued
  [2024-04-03 13:09:29] Continued by: elpaca--process
  [2024-04-03 13:09:29] Continued by: elpaca--configure-remotes
  [2024-04-03 13:09:29] Continued by: elpaca--checkout-ref
  [2024-04-03 13:09:29] Continued by: elpaca--dispatch-build-commands
  [2024-04-03 13:09:29] Queueing Dependencies
  [2024-04-03 13:09:29] Blocked by: (with-editor magit-section git-commit dash)
  [2024-04-03 13:09:29] Blocked by: (magit-section git-commit dash)
  [2024-04-03 13:09:29] Blocked by: (magit-section git-commit dash)
  [2024-04-03 13:09:29] Blocked by: (magit-section git-commit)
  [2024-04-03 13:09:29] Blocked by: (git-commit)
  [2024-04-03 13:09:29] Blocked by: (git-commit)
  [2024-04-03 13:09:29] Failed dependencies: (git-commit)
```

`M-x elpaca-info` for `git-commit`:
```
git-commit [MELPA|NonGNU-devel ELPA]
Edit Git commit messages.

source: MELPA
url: https://github.com/magit/magit
menu item recipe:
( :package "git-commit"
  :fetcher github
  :repo "magit/magit"
  :files ("lisp/git-commit.el"
	  "lisp/git-commit-pkg.el")
  :old-names (git-commit-mode)
  :source "MELPA")
full recipe:
( :package "git-commit"
  ;; Inherited from elpaca-order-functions.
  :depth 1
  :inherit t
  :protocol https
  ;; Inherited from elpaca-menu-item.
  :source "MELPA"
  :old-names (git-commit-mode)
  :files ("lisp/git-commit.el"
	  "lisp/git-commit-pkg.el")
  :repo "magit/magit"
  :fetcher github)
dependencies: 
  emacs       >= 25.1
  compat      >= 29.1.4.4
  transient   >= 20231204
  with-editor >= 20230917
dependents: 
  magit
installed version: 3.3.0.50-git b5637d6
statuses:
  (failed unblocked blocked continued-dep queued)
files:
  $REPOS/magit/lisp/git-commit-pkg.el ! $BUILDS/git-commit/git-commit-pkg.el
  $REPOS/magit/lisp/git-commit.el     ! $BUILDS/git-commit/git-commit.el
log:
  [2024-04-03 13:09:29] Package queued
  [2024-04-03 13:09:29] Continued by: elpaca--continue-dependency
  [2024-04-03 13:09:29] Continued by: elpaca--dispatch-build-commands
  [2024-04-03 13:09:29] Queueing Dependencies
  [2024-04-03 13:09:29] Continued by: elpaca--queue-dependencies
  [2024-04-03 13:09:29] Checking dependency versions
  [2024-04-03 13:09:29] elpaca--check-version: (wrong-type-argument stringp nil)
```

From the logs it looks like an issue with getting the version string, following the suggestion from the referenced issue (#265) and taking into consideration 6bbc174533b19743bf4d0728b4bd88f9e81dcb7e, evaluating:
```elisp
(elpaca-with-dir 'git-commit repo
  (elpaca-process-output "git" "log" "-n" "1" "--format=%cd" "--date=format:%Y%m%d"))
```
gives the following output:
```
"20240402
"
```
Other than the new line at the end, looks fine to me, don't know how to proceed. Thanks in advance!

**Comment by progfolio** [comment]
> Hi! I'm getting an error similar to #265 when installing magit, except for
> git-commit, all the other dependencies seems to install correctly:

Thank you very much for the detailed bug report.
It made narrowing down the issue much easier. 
This is an upstream issue in Emacs which I reported for versions 28.1 through 29.2:

https://lists.gnu.org/archive/html/bug-gnu-emacs/2024-03/msg00114.html

Looks like it's still present in 29.3 as well.

I've added to an existing workaround to cover version 29.3.
`M-x elpaca-pull` `elpaca` followed by `M-x restart-emacs` should allow git-commit's version check to complete properly. Let me know if that works for you.
